packit_service/models.py

```diff
@@ -129,9 +129,22 @@ def get_runs(self) -> List["RunModel"]:
             return trigger_list[0].runs if trigger_list else []
 
     def _get_run_item(self, model):
-        run_ids = [run.id for run in self.get_runs()]
-        with get_sa_session() as session:
-            return session.query(model).filter(RunModel.id.in_(run_ids)).all()
```

Comment:
It's some time I was playing with that. Can you please describe what actually was wrong? And ideally, create a test case for that?

Reply:
It is covered by the extension of the existing test..before the changes, after adding other copr build triggered by branch push to the db in the test, the `pr.get_copr_builds()` returned a list not only with the 1 correct copr build object, but also with this added copr build triggered by branch push, because all the run models were iterated when filtering (I assume therefore the API endpoint was dead, since this needed a lot of memory).

Reply:
When looking at it now, I am really not sure what it does and how comes it works. (There is no single `RunModel.id` property for build/test model.)

Reply:
It worked in tests only because there was only one copr build object in the DB during the test, after adding more it didn't behave correctly.